crates/uv/tests/lock.rs

```diff
@@ -697,6 +697,25 @@ fn lock_extra() -> Result<()> {
         version = "3.7.0"
         source = "registry+https://pypi.org/simple"
 
+        [[distribution]]
+        name = "project"
+        version = "0.1.0"
+        extra = "test"
```

Comment:
This demonstrates the change in format. We now have a separate distribution for `project[test]`, which includes the dependencies for both the base package and the `[test]`.

I think we could change the `project` distribution to instead include `[[distribution.optional-dependencies.test]]`? But it requires more work.